### PR TITLE
569 invoice lines taxes

### PR DESCRIPTION
**Pull request by clemens-msupply**
This makes it possible to insert and update total_before_tax, total_after_tax and tax for all invoice lines.

I removed the existing automatic calculation for total_after_tax since frontend is responsible for it.

### CODE REVIEW

File: domain/src/outbound_shipment.rs [hunk 1]
```diff
@@ -34,6 +39,9 @@ pub struct UpdateOutboundShipmentLine {
     pub item_id: Option<String>,
     pub stock_line_id: Option<String>,
     pub number_of_packs: Option<u32>,
+    pub total_before_tax: Option<f64>,
```
Comment (andreievg) [inline domain/src/outbound_shipment.rs:42]: wouldn't this be calculated from lines ?

File: domain/src/inbound_shipment.rs [hunk 2]
```diff
@@ -20,6 +22,9 @@ pub struct UpdateInboundShipment {
     pub comment: Option<String>,
     pub their_reference: Option<String>,
     pub color: Option<String>,
+    pub total_before_tax: Option<f64>,
+    pub total_after_tax: Option<f64>,
+    pub tax: Option<ShipmentTaxUpdate>,
```
Comment (andreievg) [inline domain/src/inbound_shipment.rs:27]: hmm, should this just be `taxPercentage: Option<f64>` ?

**Review by andreievg — APPROVED** [review]
Looks good, but why have totals and tax in shipment inputs ? And alos why ShipentTaxUpdate vs just f64 ?

File: domain/src/inbound_shipment.rs (hunk 2 above)
Comment (clemens-msupply) [reply domain/src/inbound_shipment.rs:27]: Sorry should have put a comment in. I need to:
1) Able to unset the tax if it is currently set -> `Some(ShipmentTaxUpdate { precentage: None})`
2) Update a tax -> `Some(ShipmentTaxUpdate { precentage: value})`
3) Keep existing tax -> `None`

A single Option can't do that...

File: domain/src/outbound_shipment.rs (hunk 1 above)
Comment (clemens-msupply) [reply domain/src/outbound_shipment.rs:42]: Removed invalid fields from non-line shipment mutations in 62c14f4

File: domain/src/inbound_shipment.rs (hunk 2 above)
Comment (andreievg) [reply domain/src/inbound_shipment.rs]: I kind of think we need a generic nullable api type for this kind of input, i've encountered this issue in a couple of places (for expiry date, batch etc.. which are optional fields in database, but in update mutation we only just have them as Optional<value>, so there is no way to set them to null). So we kind of need Option<Option<T>>

**Pull request merged**